crates/subspace-farmer/src/bin/subspace-farmer/commands/farm.rs

```diff
@@ -793,17 +795,16 @@ where
     drop(plotted_pieces);
 
     // TODO: spawn this in a dedicated thread
-    // TODO: stop if the prometheus server stops
-    let _prometheus_worker = if should_start_prometheus_server {
+    let prometheus_worker = if should_start_prometheus_server {
```

Comment:
Similarly, this neither needs a dedicated thread nor does it need to stop the whole farmer if it fails for some reason, it is a non-essential component